src/OpenTelemetry/Metrics/StringArrayEqualityComparer.cs

```diff
@@ -50,8 +54,11 @@ public bool Equals(string[] strings1, string[] strings2)
 
     public int GetHashCode(string[] strings)
     {
+        Debug.Assert(strings != null, "strings was null");
+
 #if NET6_0_OR_GREATER
         HashCode hashCode = default;
+
         for (int i = 0; i < strings.Length; i++)
```

Comment:
Use `strings!.Length` here as well?

Reply:
We could, but it isn't needed. A lot of the `!`s you see are just to make net462 happy. .NET6+ targets have better decorations across the runtime so the analysis gets smarter. In this case, we call...

```csharp
public static void Assert([DoesNotReturnIf(false)] bool condition, string? message)
```

...and the `DoesNotReturnIf` for net6.0 is what clears the warning.